Description: Fix objcopy build-id handling
 Take only .text section into account, as newer toolchains generate unique
 build ids.
 .
 Backported from GRUB 2.
Author: Lubomir Kundrak <lkundrak@redhat.com>
Forwarded: no
Last-Update: 2016-10-09

Index: b/acinclude.m4
===================================================================
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -61,7 +61,7 @@
   else
     AC_MSG_ERROR([${CC-cc} cannot link at address $link_addr])
   fi
-  if AC_TRY_COMMAND([${OBJCOPY-objcopy} -O binary conftest.exec conftest]); then :
+  if AC_TRY_COMMAND([${OBJCOPY-objcopy} --only-section=.text -O binary conftest.exec conftest]); then :
   else
     AC_MSG_ERROR([${OBJCOPY-objcopy} cannot create binary files])
   fi
Index: b/stage1/Makefile.am
===================================================================
--- a/stage1/Makefile.am
+++ b/stage1/Makefile.am
@@ -12,4 +12,4 @@
 
 SUFFIXES = .exec
 .exec:
-	$(OBJCOPY) -O binary $< $@
+	$(OBJCOPY) -O binary -R .note* -R .comment* $< $@
Index: b/stage2/Makefile.am
===================================================================
--- a/stage2/Makefile.am
+++ b/stage2/Makefile.am
@@ -276,4 +276,4 @@
 # General rule for making a raw binary.
 SUFFIXES = .exec
 .exec:
-	$(OBJCOPY) -O binary $< $@
+	$(OBJCOPY) -O binary -R .note* -R .comment* $< $@
